yt/frontends/chimera/io.py

```diff
@@ -0,0 +1,231 @@
+        pass
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+        rv = {}
+        nodal_fields = []
+        for field in fields:
+            finfo = self.ds.field_info[field]
+            nodal_flag = finfo.nodal_flag
+            if np.any(nodal_flag):
+                num_nodes = 2 ** sum(nodal_flag)
+                rv[field] = np.empty((size, num_nodes), dtype="=f8")
+                nodal_fields.append(field)
+            else:
+                rv[field] = np.empty(size, dtype="=f8")
+        ind = 0
+        for field, mesh, data in self.io_iter(chunks, fields):
+            if data is None:
+                continue
+            else:
+                ind += mesh.select(selector, data.flatten(), rv[field], ind)  # caches
+        return rv
+
+    def io_iter(self, chunks, fields):
+        for n, chunk in enumerate(chunks):
+            file = _find_files(self.filename)
+            with h5py.File(file[n], "r") as f:
+                # Generates mask according to the "ongrid_mask" variable
+                m = int(file[n][-5:-3]) - 1
+                k = f["fluid"]["entropy"].shape[0]
+                mask_0 = f["mesh"]["ongrid_mask"][k * m : k * (m + 1), :]
+
+                if f["mesh"]["array_dimensions"][2] > 1:
+                    nrd = f["mesh"]["array_dimensions"][0] - 2
+                else:
+                    nrd = f["mesh"]["array_dimensions"][0]
+
+                mask = np.repeat(mask_0[:, :, np.newaxis], nrd, axis=2).transpose()
+                for field in fields:  # Reads data by locating subheading
+                    ftype, fname = field
+                    specials = (
+                        "abar",
+                        "e_rms_1",
+                        "e_rms_2",
+                        "e_rms_3",
+                        "e_rms_4",
+                        "lumin_1",
+                        "lumin_2",
+                        "lumin_3",
+                        "lumin_4",
+                        "num_lumin_1",
+                        "num_lumin_2",
+                        "num_lumin_3",
+                        "num_lumin_4",
+                        "shock",
+                        "nse_c",
+                    )
+
+                    a_name_2 = [i.decode("utf-8") for i in f["abundance"]["a_name"]]
+                    a_name_dict = {name.strip(): name for name in a_name_2}
+                    if fname not in specials:
+                        if fname in f["fluid"]:
+                            ds = f["fluid"][f"{fname}"]
+                        elif fname in f["abundance"]:
+                            ds = f["abundance"][f"{fname}"]
+                        elif fname in a_name_dict:
+                            ind_xn = a_name_2.index(a_name_dict[fname])
+                            ds = f["abundance"]["xn_c"][:, :, :, ind_xn]
+                        else:
+                            sys.exit("Error: Invalid field name")
```

Comment:
Actually I'm not even sure: do we want to this to be unrecoverable, or would a warning suffice ? And in any case, it'd be better to print out the problematic `fname` in the message
```suggestion
                            warning.warn(f"Invalid field name {fname}")
```

Reply:
Note -- I also commented on this, but resolved my conservation so we're not discussing it in different comments :) I had suggested raising an error instead of the `sys.exit`. A warning is fine by me as well.

Reply:
Sorry I intended this to be a reply to your comment, I meant to say "I'm not even sure *that this should be an error*".
I think it's sane to have redundant/extra validation, since the model is quite complicated.